docs/graphql/core/guides/environments.rst

```diff
@@ -0,0 +1,268 @@
+After that, we can initialise a Hasura project by running the :ref:`hasura init <hasura_init>` command which will scaffold a project directory with migrations, metadata and config.
+
+*New database*: If your database is clean without any existing schema, you can start using the console via the CLI (:ref:`hasura console <hasura_console>`), 
+modify the Postgres schema, and the CLI will take care of creating the ``up`` and ``down`` migration files.
+
+*Existing database*: In case you have an existing database schema, you can use the CLI to initialise the migration for that schema using the following command:
+
+.. code-block:: bash
+
+    hasura migrate create init-schema --from-server
+
+This will take a ``pg_dump`` of the ``public`` schema and create an ``up`` migration to get started. 
+
+Alternatively, if you have an SQL file with all the DDL statements, you can also specify that as an argument:
+
+.. code-block:: bash
+
+    hasura migrate create init-schema --sql-from-file ./file.sql
+
+Once you set this up, you can continue to use the Hasura console served via the CLI and make any schema changes and migration files will be automatically created as you work along.
+
+.. note::
+
+    If you're interested in a more extensive guide on setting up migrations, see :ref:`this docs page <migrations_setup>`.
+
+Squashing migrations
+********************
+
+During local development, we typically iterate over schema modifications multiple times. 
+This leads to a large number of migration files being created over time. 
+If you are building a schema for a specific feature and you really don't want to roll back or manage the smaller iterations in between, 
+you can :ref:`squash the migration files into a single file <hasura_migrate_squash>` for easier organisation:
+
+.. code-block:: bash
+
+    hasura migrate squash --from <version>
+
+
+Metadata
+^^^^^^^^
+
+The state of Hasura metadata is managed via snapshots of the metadata. Hasura stores this metadata to create the GraphQL API over Postgres and provide other functionalities like remote schemas, event triggers etc. 
+All the actions performed on the console, like tracking tables/views/functions, creating relationships, configuring permissions, 
+creating event triggers and remote schemas, etc. can be exported as a JSON/yaml metadata file.
+
+These snapshots can be :ref:`exported <export_hasura_metadata>` and :ref:`imported <import_hasura_metadata>` as a whole to configure Hasura to a state represented in the snapshot.
+
+The metadata directory of your Hasura project should be included in your version control system like git, 
+so that you can roll back corresponding changes later, if required.
+
+.. note::
+
+    Read more about Hasura metadata on :ref:`this docs page <manage_hasura_metadata>`.
+
+Developing and testing business logic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Hasura lets you write business logic in a flexible way. 
```

Comment:
This can be re-written as **Hasura allows you to write business logic in a flexible way.**